connectivity/suite.go

```diff
@@ -240,6 +254,74 @@ func Run(ctx context.Context, ct *check.ConnectivityTest) error {
 			return check.ResultDrop, check.ResultNone
 		})
 
+	// Tests with deny policy
+	if ct.DenyPolicy() {
+		// This policy denies ingress into client from client2.
+		ct.NewTest("client-ingress-deny").WithPolicy(clientIngressFromClient2DenyPolicyYAML).
+			WithScenarios(
+				tests.ClientToClient(),
+			).
+			WithExpectations(func(a *check.Action) (egress, ingress check.Result) {
+				if a.Destination().HasLabel("kind", "client") {
+					return check.ResultOK, check.ResultDrop
+				}
+				return check.ResultOK, check.ResultOK
+			})
+
+		// This policy allows ingress to echo only from client with a label 'other:client'
+		ct.NewTest("echo-ingress-deny").WithPolicy(echoIngressFromOtherClientDenyPolicyYAML).
+			WithScenarios(
+				tests.PodToPod(),
+			).
+			WithExpectations(func(a *check.Action) (egress, ingress check.Result) {
+				if a.Destination().HasLabel("kind", "echo") {
+					return check.ResultDrop, check.ResultDrop
+				}
+				return check.ResultOK, check.ResultOK
+			})
+
+		// This policy denies ingress to echo only from client with a label 'other:client'.
+		ct.NewTest("client-ingress-icmp-deny").WithPolicy(echoIngressICMPDenyPolicyYAML).
+			WithFeatureRequirements(check.RequireFeatureEnabled(check.FeatureICMPPolicy)).
+			WithScenarios(
+				tests.ClientToClient(),
```

Comment:
Would be nice to also have non-ICMP traffic to validate the difference in policy behavior, if any?